packages/web-twig/src/Resources/components/Grid/Grid.twig

```diff
@@ -19,11 +22,51 @@
 {%- endif -%}
 
 {%- set _classNames = [ _rootClassName, _styleProps.className ] | merge(_colsClasses) -%}
+{%- set _style = '' -%}
+
+{%- if _spacing is iterable -%}
+    {%- for breakpoint, breakpointValue in _spacing -%}
+        {%- set suffix = (breakpoint == 'mobile') ? '' : '-' ~ breakpoint -%}
+        {%- set _style =
+            _style ~
+                '--grid-spacing-x' ~ suffix ~ ': var(--spirit-' ~ breakpointValue ~ ');
+                --grid-spacing-y' ~ suffix ~ ': var(--spirit-' ~ breakpointValue ~ ');'
+        -%}
+    {%- endfor -%}
+{%- elseif _spacing -%}
+    {%- set _style =
+        _style ~
+            '--grid-spacing-x: var(--spirit-' ~ _spacing ~ ');
+            --grid-spacing-y: var(--spirit-' ~ _spacing ~ ');'
+    -%}
+{%- endif -%}
+
+{%- if _spacingX is iterable -%}
+    {%- for breakpoint, breakpointValue in _spacingX -%}
+        {%- set suffix = (breakpoint == 'mobile') ? '' : '-' ~ breakpoint -%}
+        {%- set _style = _style ~ '--grid-spacing-x' ~ suffix ~ ': var(--spirit-' ~ breakpointValue ~ ');' -%}
+    {%- endfor -%}
+{%- elseif _spacingX -%}
+    {%- set _style = _style ~ '--grid-spacing-x: var(--spirit-' ~ _spacingX ~ ');' -%}
+{%- endif -%}
+
+{%- if _spacingY is iterable -%}
+    {%- for breakpoint, breakpointValue in _spacingY -%}
```

Comment:
I wonder if the lines inside `if` and `elseif` can be abstracted, there is a lot of repeated code…

```twig
{%- if _spacing is iterable -%}
  {# set_responsive_stuff(x) #}
  {# set_responsive_stuff(y) #}
{%- elseif _spacingX -%}
  {# set_single_value(x) #}
  {# set_single_value(y) #}
{%- endif -%}

{%- if _spacingX is iterable -%}
  {# set_responsive_stuff(x) #}
{%- elseif _spacingX -%}
  {# set_single_value(x) #}
{%- endif -%}

{# … #}
```

Reply:
Hope you don't mind but I skip this comment, because I'm not sure how to implement this abstraction in Twig.

But maybe we could take a look at it in https://jira.almacareer.tech/browse/DS-1423?

Reply:
Yes, I'm completely fine with that.